getting_started/setup_linux/prepare_build_environment.rst

```diff
@@ -0,0 +1,104 @@
+.. _prepare-build-environment:
+
+Set up a development environment to build the kernel
+####################################################
+
+.. contents::
+   :local:
+   :depth: 3
+
+Introduction
+************
+
+
+These instructions will help you set up a development environment for the SOF branch of the Linux kernel. If you have dedicated test hardware you can use ktest to install it over ssh, otherwise you can install it locally on your device in addition to your default kernel.
+
+Prerequisites
+*************
+
+1. Device requirements
+-----------------------
+
+**Development device:** PC running Fedora 35+ or Ubuntu 20.04+.
+
+**Target device:** PC running Fedora 35+ or Ubuntu 20.04+, with secure boot disabled. If the target device is different than the development device, they must be on the same local network so you can ssh into the target.
+
+2. Create working dir
+---------------------
+
+This can be located anywhere, but later commands assume its path is ~/sof
+It will host all the sources you need to get
+
+.. code-block:: bash
+   
+   mkdir ~/sof
+   cd ~/sof
+
+3. Install kernel build dependencies
+------------------------------------
+
+Fedora:
+
+.. code-block:: bash
+
+   sudo dnf install fedpkg
+   fedpkg clone -a kernel
```

Comment:
I really don't see why `fedpkg` and `dnf builddep kernel.spec` are necessary. These are meant to help with RPM packages, yet the Ubuntu section has no `dpkg` equivalent.

Reply:
[Fedora's kernel building instructions](https://docs.fedoraproject.org/en-US/quick-docs/kernel/build-custom-kernel/#_get_the_dependencies) only mention the fedpkg way. [Ubuntu's docs](https://wiki.ubuntu.com/Kernel/BuildYourOwnKernel) are a bit weirder. They say that the `apt build-dep` is required in addition to the list of deps, yet I was able to build the kernel just fine without the `apt build-dep`. Since it wasn't necessary for me on a clean install, I decided to leave that out and only put the list of deps.

Also, I think the fedpkg gets deps other than for building rpms (ie zstd). I was not able to build the kernel with make before running the fedpkg command. There may be a list of deps you can install instead of using fedpkg, but I just went with what Fedora's docs say.